Ignore false positives in the unused dependencies report

irfano opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This excludes false positive dependencies from unused dependencies report.

For `libs.squareup.okhttp3.tls`, the project can still run without it, but it was [added recently](https://github.com/woocommerce/woocommerce-android/pull/14681) because the card connection flow crashes without it. So, I added an exception for it as well.

### Test Steps
<!-- Describe how to test your changes. Include only what’s needed for the reviewer to validate the behavior.
For new features, outline the main user flow or goal rather than every tap or click — the reviewer should be able to complete the flow naturally.
If applicable, mention key devices, scenarios, or edge cases to verify. -->
1. Run `./gradlew buildHealth `
2. Find the line in the terminal logs that says.
3. Click on it.
4. Verify that no unused dependencies are reported.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

irfano committed:
Remove apache.http.client.android dependency

Removes the `apache.http.client.android` dependency from `androidTestImplementation` configurations.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

irfano committed:
Revert "Remove apache.http.client.android dependency"

This reverts commit 895ff5c391051a136a8aa6e180173f528f2187be.

irfano committed:
Configure dependency analysis to ignore false positives

irfano committed:
Exclude unused dependencies from analysis for Woocommerce-Wear

Configures the dependency analysis plugin to ignore certain dependencies that are reported as unused but are required at runtime.

irfano committed:
Exclude Unused Dependencies from Dependency Analysis for apifaker

Exclude `androidx.compose.material.icons.extended` and `androidx.hilt.navigation.compose` from the dependency analysis for the `apifaker` library.

irfano committed:
Exclude androidx.core.ktx from dependency analysis for fluxc-plugin

This dependency is required for the app to build successfully, so it's being excluded from the unused dependency check to prevent build failures.

irfano committed:
Exclude androidx.paging.runtime from unused dependency check

Adds a `dependencyAnalysis` rule to `fluxc-tests` to exclude `androidx.paging.runtime`.

irfano committed:
Exclude okhttp3-tls from dependency analysis

This dependency is required to prevent a `NoClassDefFoundError` during the card reader connection flow.

irfano commented:
Sorry for not adding more information to the PR description. I'm monitoring the Dependency Analysis Alert (p1762837882059759-slack-C8ZFBDQC9) as part of my exception catcher rotation, and these false positives appeared in the health report.
I followed the second suggestion in the guide (paqN3M-1d0-p2) to suppress these false positives.

AdamGrzybkowski reviewed (approved):
The build appears to be healthy now. Do we know what has changed recently that caused so many false positives?

irfano commented:
> Do we know what has changed recently that caused so many false positives?

These were probably in the report for a long time, but we just hadn’t taken action before.
